The upstream patch would not apply cleanly, so perform the same changes.

Original patch header:
From 00148329967adb196138372771052a3f606a6ea3 Mon Sep 17 00:00:00 2001
From: coypu <coypu@sdf.org>
Date: Wed, 2 Mar 2016 19:43:10 +0200
Subject: [PATCH 2/2] gdate: Suppress string format literal warning

Newer versions of GCC emit an error here, but we know it's safe.
https://bugzilla.gnome.org/761550

Upstream-Status: Backport

Signed-off-by: Tim Orling <timothy.t.orling@linux.intel.com>

Index: git/glib/glib/gdate.c
===================================================================
--- git.orig/glib/glib/gdate.c
+++ git/glib/glib/gdate.c
@@ -2439,6 +2439,8 @@ win32_strftime_helper (const GDate     *
  *
  * Returns: number of characters written to the buffer, or 0 the buffer was too small
  */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
 gsize     
 g_date_strftime (gchar       *s, 
                  gsize        slen, 
@@ -2549,3 +2551,5 @@ g_date_strftime (gchar       *s,
   return retval;
 #endif
 }
+
+#pragma GCC diagnostic pop
